vuepress/docs/next/tutorials/consume/external-db-ms.md

```diff
@@ -0,0 +1,203 @@
+ ```
+ 
+2. Create the JSON plugin configuration Secret for the microservice plugin with a name that reflects the bundle ID and plugin code as Kuternetes is not tenant aware.
+
+   The name should begin with "pn" plus the concatenation of the bundle ID, plugin code, and microservice name, connected by dashes. The name can be appended with suffixes such as "SECRET" for better identification.  
+
+   E.g.  
+   Bundle ID = `hasldk12`, Plugin Code = `8dsjahj2`, Microservice name = YOUR-MS-NAME  
+   Your Configuration Secret= `pn-hasldk12-8dsjahj2-YOUR-MS-NAME-SECRET`
+
+3. Add the following env vars per tenant. The key for your primary tenant must be 'primary' regardless of what the actual tenant ID is. Secondary tenant keys should refer to their tenant IDs. Add the key-value pairs for the Spring Datasource Username, password, URL, and connection driver, in a base 64 encoded format.
+
+This is a decrypted example of a plugin configuration Secret:
+``` json
+apiVersion: v1
+kind: Secret
+metadata:
+  name: pn-hasldk12-8dsjahj2-YOUR-MS-NAME-SECRET
+  namespace: YOUR-NAMESPACE
+type: Opaque
+data:
+  primary: >-
+    {
+  "environment_variables": [
+    {
+      "name": "SPRING_DATASOURCE_USERNAME",
+      "valueFrom": {
+        "secretKeyRef": {
+          "name": "hasldk12-8dsjahj2-YOUR-SECRET-PRIMARY",
+          "key": "YOUR-USERNAME"
+        }
+      }
+    },
+    {
+      "name": "SPRING_DATASOURCE_PASSWORD",
+      "valueFrom": {
+        "secretKeyRef": {
+          "name": "hasldk12-8dsjahj2-YOUR-SECRET-PRIMARY",
+          "key": "YOUR-PASSWORD"
+        }
+      }
+    },
+    {
+      "name": "SPRING_DATASOURCE_URL",
+      "valueFrom": {
+        "secretKeyRef": {
+          "name": "hasldk12-8dsjahj2-aYOUR-SECRET-PRIMARY",
+          "key": "YOUR-URL"
+        }
+      }
+    },
+    {
+      "name": "SPRING_JPA_DATABASE_PLATFORM",
+      "value": "org.hibernate.dialect.PostgreSQLDialect"
+    }
+  ]
+}
+  tenant1: >-
+    {
+  "environment_variables": [
+    {
+      "name": "SPRING_DATASOURCE_USERNAME",
+      "valueFrom": {
+        "secretKeyRef": {
+          "name": "hasldk12-8dsjahj2-YOUR-SECRET-TENANT1",
+          "key": "YOUR-USERNAME"
+        }
+      }
+    },
+    {
+      "name": "SPRING_DATASOURCE_PASSWORD",
+      "valueFrom": {
+        "secretKeyRef": {
+          "name": "hasldk12-8dsjahj2-YOUR-SECRET-TENANT1",
+          "key": "YOUR-PASSWORD"
+        }
+      }
+    },
+    {
+      "name": "SPRING_DATASOURCE_URL",
+      "valueFrom": {
+        "secretKeyRef": {
+          "name": "hasldk12-8dsjahj2-YOUR-SECRET-TENANT1",
+          "key": "YOUR-URL"
+        }
+      }
+    },
+    {
+      "name": "SPRING_JPA_DATABASE_PLATFORM",
+      "value": "org.hibernate.dialect.PostgreSQLDialect"
+    }
+  ]
+}
+```
+The `secretKeyRef` `name` must also follow the same convention as the plugin configuration Secret, with the bundle ID and plugin code concatenated to the tenant reference name with dashes.
```

Comment:
Here too: bundle ID + plugin ID or simply plugin code

Reply:
@jyunmitch the `pn-` prefix is the official way. I've found an entry point to use it also without `pn-` in some edge cases, but let's keep the official one